src/plotly/api.nim

```diff
@@ -19,13 +18,47 @@ type
     Left = "left"
     Right = "right"
 
+  ErrorBarKind = enum   # different error bar kinds (from constant value, array,...)
```

Comment:
I think this should be exported so the user can construct from scratch. I would also prefer to use a `{.pure.}` enum and use human readable names like  `ErrorBarKind.ConstantSymmetric` instead of `ebkConstantSym`. What do you think?

Reply:
I kept it private, because I mainly wanted that `enum` to be used for the object variant of `ErrorBar`. The prefix is then simply to clarify it's an `enum` value, following the Nim style guide here: 

https://nim-lang.org/docs/nep1.html

I suppose pure `enum`s can be used for object variants too, if one wants to export the symbol (in which case I agree that your naming convention is better suited!).

Reply:
but if it is private, the user won't be able to construct a plot with error bars without using your helpers, is that correct?

Reply:
I guess that is right, yes. My thought process was more along the lines of abstracting the "ugliness" of dealing with the `ErrorBarKind` away. Since the `ErrorBar` is an object variant creating such an object in the way one does for e.g. `Trace` is bound to be cumbersome. The user would need to know about the different names of the fields for each object variant etc.

We can export the `enum`, that's fine with me. But the goal should rather be to provide an even nicer interface (than it is right now using `newErrorBar`), which unifies the different kinds rather than expecting the user to work on that low level, no?

Reply:
Ideally there would be a way to allow for handing the `Trace` simply:
- a constant error value
- a high and a low value
- a seq matching the data seq
- 2 seqs matching the data seqs

without any need to create a full `ErrorBar` object on the user side, I'd say. In that case the error bars would inherit the colors from the marker for example.

One could additionally introduce a `setXErrorBar(tr: Trace, ...)` proc, which can then e.g. be called
```nim
d.setXErrorBar(1.5)
d.setXErrorBar((0.5, 1.5))
d.setXErrorBar(@[seq of # elements of data])
d.setXErrorBar((seq1, seq2))
```
which does just that maybe? These could also get arguments for percent. 

Not sure what the nicest interface (for the most general cases anyways) is right now.

Reply:
> But the goal should rather be to provide an even nicer interface (than it is right now using newErrorBar
I agree completely. 

I also agree we can figure out the API for error bars to make it more ergonomic. I think that could be after this PR so we have a chance to see how other tools do it.

Reply:
Agreed!